samples/nrf_desktop/src/modules/usb_state.c

```diff
@@ -387,6 +428,17 @@ static bool event_handler(const struct event_header *eh)
 
 			return false;
 		}
+
+		if (is_config_fetch_event(eh)) {
+			struct config_fetch_event *event = cast_config_fetch_event(eh);
+
+			__ASSERT_NO_MSG(event->dyndata.size <= sizeof(fetched_data));
```

Comment:
if fetched data size comes from outside then assert must not be used

Reply:
it comes from other part of firmware and should only catch programming mistakes